eva/storage/sqlite_storage_engine.py

```diff
@@ -196,22 +198,14 @@ def delete(self, table: TableCatalogEntry, where_clause: Dict[str, Any]):
         """
         try:
             table_to_delete_from = self._try_loading_table_via_reflection(table.name)
-            table_columns = [
-                col.name
-                for col in table_to_delete_from.columns
-                if col.name != "_row_id"
-            ]
-            filter_clause = []
+
+            filter_clause = predicate_node_to_filter_clause(
+                table=table_to_delete_from, predicate_node=where_clause
+            )
```

Comment:
Nit: I feel it is better to move `predicate_node_to_filter_clause` logic into `delete_executor` and the `delete_executor` accepts the `ColumnElement[bool]` as the input argument (i.e., where_clause).  `ColumnElement[bool]` is the return type of `and_` and `or_`

Reply:
Good idea. Moved it to `delete_executor`